### ISSUE TITLE
Tensor.register_hook is not passing the tensor object to the hook function

### ISSUE THREAD

**Issue by timmeinhardt** [pytorch/pytorch#10683]
I am currently working with `Tensor.register_hook` and I want to manipulate the gradients of a parameter with respect to the values of its data, i.e., the weights or bias itself. This would be very easy if `self` is also passed to the hook function such as in `Module.register_backward_hook `. Is there a reason why this is not done? This could be achieved by adding:
```
wrapper = functools.partial(hook, self)
functools.update_wrapper(wrapper, hook)
```
to the `register_hook` method.

**Comment by zou3519** [comment]
This sounds reasonable, but it might not be good for bc.

**Comment by zou3519** [comment]
Yes, backwards compatibility. That's reasonable as long as it is feasible to do with python
